capnpc-go/capnpc-go.go

```diff
@@ -966,14 +966,27 @@ func (t Type) json(w io.Writer) {
 func (n *node) defineNewStructFunc(w io.Writer) {
 	assert(n.Which() == NODE_STRUCT, "invalid struct node")
 
-	fprintf(w, "func New%s(s *C.Segment) %s { return %s(s.NewStruct(%d, %d)) }\n",
-		n.name, n.name, n.name, n.Struct().DataWordCount()*8, n.Struct().PointerCount())
-	fprintf(w, "func NewRoot%s(s *C.Segment) %s { return %s(s.NewRootStruct(%d, %d)) }\n",
-		n.name, n.name, n.name, n.Struct().DataWordCount()*8, n.Struct().PointerCount())
-	fprintf(w, "func AutoNew%s(s *C.Segment) %s { return %s(s.NewStructAR(%d, %d)) }\n",
-		n.name, n.name, n.name, n.Struct().DataWordCount()*8, n.Struct().PointerCount())
-	fprintf(w, "func ReadRoot%s(s *C.Segment) %s { return %s(s.Root(0).ToStruct()) }\n",
-		n.name, n.name, n.name)
+	var (
+		datasz = n.Struct().DataWordCount() * 8
+		ptrs   = n.Struct().PointerCount()
+	)
+	if !disabledStructFunc {
+		fprintf(w, "func New%s(s *C.Segment) %s { return %s(s.NewStruct(%d, %d)) }\n",
+			n.name, n.name, n.name, datasz, ptrs)
+		fprintf(w, "func NewRoot%s(s *C.Segment) %s { return %s(s.NewRootStruct(%d, %d)) }\n",
+			n.name, n.name, n.name, datasz, ptrs)
+		fprintf(w, "func AutoNew%s(s *C.Segment) %s { return %s(s.NewStructAR(%d, %d)) }\n",
+			n.name, n.name, n.name, datasz, ptrs)
+		fprintf(w, "func ReadRoot%s(s *C.Segment) %s { return %s(s.Root(0).ToStruct()) }\n",
+			n.name, n.name, n.name)
+	}
+
+	if enabledStructInfo {
+		fprintf(w, "func (s *%s) StructInfo() (int, int) { return %d, %d }\n",
+			n.name, datasz, ptrs)
+		fprintf(w, "func (s *%s) NewP(p C.Struct) *%s {\nv := %s(p)\nreturn &v\n}\n",
```

Comment:
```suggestion
		fprintf(w, "func (s *%s) NewP(p C.Struct) *%s {\n", n.name, n.name)
		fprintf(w, "\tv := %s(p)\n", n.name)
		fprintf(w, "\treturn &v\n")
		fprintf(w, "}\n")
```
인덴테이션이 추가되면 좋을 것 같고, 아래 다른 코드를 보니 일반적으로 멀티라인은 fprintf를 여러 개 사용해서 구분하는 듯 합니다

https://github.com/devsisters/gb-server-game/pull/4087/commits/b629537019edfaf54e14404ddf6119ade0c0a051#diff-41442fa825075d5d4e99f853a372de1c58c745d7ec6f2d55c2ad296d53cae377R21-R24
그나저나 여기선 따로 인덴테이션을 안 해주고 있는 것 같은데, 게임서버 formatter가 자동으로 해줬나보네요

Reply:
https://github.com/devsisters/go-capnproto/pull/3/commits/5d0a62d6c4e9adfa1b60d53ca08dd0bd4e943bcb
\+ 커밋이 추가되면서 게임서버 PR 쪽에도 [diff](https://github.com/devsisters/gb-server-game/compare/e1224b0d045612afa72da13cf61c42e9f1f3f66f..338ce6f07cdba6fa3fe4d35100db427cb76a8c2c)가 생겼으니 참고 부탁드립니다